core/tests/webdriverio_utils/users.js

```diff
@@ -0,0 +1,214 @@
+    // Click the "sign in" button to trigger redirection.
+    await action.click('Sign in button', signInButton);
+  }, (url) => {
+    // Users will be redirected to preferred dashboard if they are fully
+    // registered. Otherwise, they will be redirected to signup page.
+    // eslint-disable-next-line max-len
+    return /(learner-dashboard|creator-dashboard|signup|pending-account-deletion)/.test(url);
+  }, async() => {
+    // Cannot predict the new page, so waiting for loading message to disappear.
+    await waitFor.pageToFullyLoad();
+  });
+};
+
+var logout = async function() {
+  await waitFor.clientSideRedirection(async() => {
+    await browser.url(general.SERVER_URL_PREFIX + general.LOGOUT_URL_SUFFIX);
+  }, (url) => {
+    // Wait until the URL has changed to something that is not /logout.
+    return !(/logout/.test(url));
+  }, async() => {
+    await waitFor.visibilityOf(
+      $(splashPage), 'Splash page takes too long to appear');
+    await waitFor.pageToFullyLoad();
+  });
+};
+
+// The user needs to log in immediately before this method is called. Note
+// that this will fail if the user already has a username.
+var _completeSignup = async function(username) {
+  await waitFor.pageToFullyLoad();
+  var cookieBannerAcceptButton = $(
+    '.e2e-test-oppia-cookie-banner-accept-button');
+  var cookieButtonPresent = await cookieBannerAcceptButton.isDisplayed();
+  if (cookieButtonPresent) {
+    await action.click('Accept Cookie Button', cookieBannerAcceptButton);
+  }
+
+  var signupPage = $('.e2e-test-signup-page');
+  await waitFor.presenceOf(signupPage, 'Signup page did not load');
+
+  var usernameInput = $('.e2e-test-username-input');
+  await action.keys('Username input', usernameInput, username);
+
+  var agreeToTermsCheckbox = await (
+    $('.e2e-test-agree-to-terms-checkbox'));
+  await action.click('Agree to terms checkbox', agreeToTermsCheckbox);
+
+  var registerUser = $('.e2e-test-register-user');
+
+  var currentUrl = decodeURIComponent(await browser.getUrl());
+
+  var returnUrl = currentUrl.split('return_url=')[
+    currentUrl.split('return_url=').length - 1];
+
+  await waitFor.clientSideRedirection(async() => {
+    // Click the "register user" button to trigger redirection.
+    await action.click('Register user button', registerUser);
+  }, (url) => {
+    if (returnUrl === '/') {
+      return /(learner-dashboard|creator-dashboard)/.test(url);
+    } else {
+      return (new RegExp(returnUrl)).test(url);
+    }
+  }, async() => {
+    // Cannot predict the new page, so waiting for loading message to disappear.
+    await waitFor.pageToFullyLoad();
+  });
+};
+
+var createAndLoginUser = async function(
+    email, username, useManualNavigation = true) {
+  await _createFirebaseAccount(email);
+  await login(email, useManualNavigation);
+  await _completeSignup(username);
+};
+
+var createAndLoginCurriculumAdminUser = async function(email, username) {
+  var adminPage = new AdminPage.AdminPage();
```

Comment:
No, we cannot declare this at the top as this will throw error "$ is not defined" in Adminpage.js. This was the fix for being able to declare variable at top in AminPage.js

Reply:
I would say add a comment above wherever you have used this so that no one breaks it.

Reply:
Done.